### ISSUE TITLE
Rear light

### ISSUE THREAD

**Issue by j4cobl** [Imaginous/Cowboy_Untamed#37]
Hello, thank for your great app!

Since you know the bike so well do you have any diagram or stuff like that about the battery connector?

Rear light never worked, I planned to open it some day.

Thank you

**Comment by Imaginous** [comment]
It's connected to the data bus of the bike. It's turned on/off via commands.

Seen the problem before, it's a soldering failure in the battery (most of the time).

Get a battery replacement under warranty.

Note: you have to send back the defective battery.

**Comment by philffm** [comment]
@j4cobl 

Have you found schematics or something? My C2 Battery (2019) also shows some weird behavior. Light does rarely ever work at 100% charge but after a few kilometers cycling and around 80% charge it shows a weak red (flickering) light. 

Unfortunately there are no teardown videos around yet  - have you opened the battery in the meantime? I am wondering how to access the light the best way possible without messing too much with the actual battery.

**Comment by jpbourhis29** [comment]
Hello @philffm,

I already open the battery to fix it by my self ( I work for a company specialized on EV and hybrid vehicles) I got a trouble on a 0 Voltage battery  (which was the value read by the BMS and not the cells values, cells values was good I test them independently) and I was able to fix the problems in the electronics BMS parts.
And I got the same problem with the rear light my bike was able to drive but the light was not working. I disassemble the battery, test the rear light independantly and it was working well, the problem was a bad solder and there was some trace of oxidation , by cleaning the pcb parts and remaking the soldering it's now work perfectly.

**Comment by philffm** [comment]
Do you remember how you got the battery out? Is it only the silicone glue in the bottom? Currently trying it with a friend

**Comment by jpbourhis29** [comment]
Yes I remember well, not able to got some picture for you, I don't have my bike and battery right now.
But to give you some explanation you need to remove 4 screws on the botom of the battery and remove the cover then you will see the cell (the battery will slide by this opening)
but there is still cable connected to the battery, you need to remove the rear light, it's insert inside the metal casing by just some clip and a joint arround,  you need to find something really thin to pop the light out.
and be carefull the cable will be connected you can unplug. Then you need to unplug the battery voltage and data connector just behind the light. and finally you can slide the battery pack outside the casing it's litlle bit hard,  I use a strong plastic piece ( not metallic of course) to start pushing the battery from the light hole and the grab from the bottom and you will have access to BMS and cells.

**Comment by philffm** [comment]
![image](https://github.com/Imaginous/Cowboy_Untamed/assets/6079545/034b208d-7077-4f7d-a260-ec93552a150b)![image](https://github.com/Imaginous/Cowboy_Untamed/assets/6079545/20e2feb8-988c-46f6-9425-e871db8d5d4c)![image](https://github.com/Imaginous/Cowboy_Untamed/assets/6079545/78036fa7-07c1-4a78-bd70-0a3f2e9a228c)

Thanks for your instructions :) really helped!

Still not working rear light though. Do you remember if the PCB looked the same? D2 seems missing 🤷🏽‍♂️

**Comment by j4cobl** [comment]
Does it look like this when you open the bottom? The battery cells doesnt come out.

I removed the 20 inches long scew in the middle but doesn't move either.

Thank for your help guys.


![PXL_20231006_223527601.jpg](https://github.com/Imaginous/Cowboy_Untamed/assets/75327489/342f9c1b-4812-4311-90b5-22a5cf66f174)

**Comment by philffm** [comment]
> Does it look like this when you open the bottom? The battery cells doesnt come out.
> 
> I removed the 20 inches long scew in the middle but doesn't move either.
> 
> Thank for your help guys.
> 
> 
> ![PXL_20231006_223527601.jpg](https://github.com/Imaginous/Cowboy_Untamed/assets/75327489/342f9c1b-4812-4311-90b5-22a5cf66f174)
> 
> 

Exactly - the clean approach is to first unscrew the 4 screws of the charging port / pry open the rear light (just the transparent lid, not the black surrounding plastic - this doesn't move either), disconnect it from the top part and then the battery should be able to slide out like charm
